### DISCUSSION TITLE
Hikvision (DS-KD8003) ISAPI command : Open a door / Relay

### DISCUSSION THREAD

**Discussion by ayghazali** [Sunoo/homebridge-camera-ffmpeg#1145] [Ideas]
Can someone add the function of open door/relay for hikvision doorbells?

I found this [home assistant thread ](https://community.home-assistant.io/t/ds-kd8003-ds-kv8113-ds-kv8213-ds-kv6113-ds-kv8413-and-integration-hikvision-hikconnect-video-intercom-doorbell/238535) discussing this and they had this command for open door

To open an electrical door (relay 1 or 2)
curl -i --digest -u admin:xxx -X PUT -d '<RemoteControlDoor><cmd>open</cmd></RemoteControlDoor>' http://192.168.0.xx/ISAPI/AccessControl/RemoteControl/door/1

Not sure if this can be implemented into this plug-in or any other plugin. 

Apologies for the noob question, I'm not a coder, so I don't know how these things work exactly

**Comment by Sunoo** [comment]
I don’t think it necessarily makes sense to add to this plugin. Someone could definitely make it a separate plugin, however, as you already have a command, you should be able to add that to homebridge-cmd4. I believe there are also plugins specifically for making HTTP requests, but I’ve never used them, and have no idea if they support PUT.
